File: frontend/api/run_migration.py
import os
import psycopg2
from dotenv import load_dotenv

# Load variables from .env
load_dotenv()

db_url = os.environ.get("DATABASE_URL")

if not db_url:
    print("Error: DATABASE_URL environment variable is not set.")
    exit(1)

migration_path = "../supabase/migrations/20260709000001_chat_memory.sql"

print(f"Connecting to {host}...")
try:
    conn = psycopg2.connect(db_url)
    conn.autocommit = True
    cursor = conn.cursor()
    
    with open(migration_path, "r") as f:
        sql = f.read()
    
    print("Executing migration...")
    cursor.execute(sql)
    print("Migration executed successfully.")
    
except Exception as e:
    print(f"Error executing migration: {e}")
finally:
    if 'conn' in locals() and conn:
        conn.close()

File: apps/admin/run_migration.mjs
import { Client } from 'pg';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';
import dotenv from 'dotenv';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// Load .env variables
dotenv.config({ path: path.join(__dirname, '.env.local') });

const connectionString = process.env.DATABASE_URL;

if (!connectionString) {
  console.error("Error: DATABASE_URL is not set in the environment.");
  process.exit(1);
}

async function main() {
  const client = new Client({
    connectionString,
    ssl: { rejectUnauthorized: false }
  });

  try {
    console.log("Connecting to Supabase...");
    await client.connect();
    
    console.log("Reading migration file...");
    const sql = fs.readFileSync(path.join(__dirname, '../supabase/migrations/20260709000000_init.sql'), 'utf-8');
    
    console.log("Executing migration...");
    await client.query(sql);
    console.log("Migration executed successfully!");
    
  } catch (error) {
    console.error("Migration failed:", error);
  } finally {
    await client.end();
  }
}

main();
